trl/models/modeling_base.py

```diff
@@ -108,3 +112,69 @@ def save_pretrained(self, *args, **kwargs):
         Save the pretrained model to a directory.
         """
         return self.pretrained_model.save_pretrained(*args, **kwargs)
+
+
+def create_reference_model(model: PreTrainedModelWrapper, share_layers: int = None, pattern: str = None):
+    """
+    Creates a static reference copy of a model. Note that model will be in `.eval()` mode.
+    
+    Args:
+        model (`PreTrainedModelWrapper`): model to be copied
+        share_layers (`int`, *optional*): first layers are shared between both models and kept frozen. 
+        pattern (`str`, *optional*): The shared layers are selected with a string pattern 
+            (e.g. "transformer.h.{layer}" for GPT2) and if a custom pattern is necessary it can be passed here.
+
+    Returns
+        `PreTrainedModelWrapper`
+    """
+
+    parameter_names = [n for n, _ in model.named_parameters()]
+    ref_model = deepcopy(model)
```

Comment:
Do you think this can blow up the memory if manipulating very large models?
I was thinking of leveraging `accelerate.init_empty_weights` context manager and initialize an empty model and populating it step by step. This can be boilerplate a bit so maybe let's leave it in a follow up PR

Reply:
Sounds good - let's see how it works and maybe adapt it to `init_empty_weights` in a follow up :)